pppd/ipcp.c

```diff
@@ -196,15 +196,6 @@ static struct option ipcp_option_list[] = {
       "disable defaultroute option", OPT_ALIAS | OPT_A2CLR,
       &ipcp_wantoptions[0].default_route },
 
-#ifdef __linux__
-    { "replacedefaultroute", o_bool,
-				&ipcp_wantoptions[0].replace_default_route,
-      "Replace default route", OPT_PRIV | 1
-    },
-    { "noreplacedefaultroute", o_bool,
-				&ipcp_wantoptions[0].replace_default_route,
-      "Do not replace default route", 0 },
-#endif
```

Comment:
In general I prefer not to break existing working setups by removing options, if it is reasonable for the option to become a no-op (or at worst emit a warning). Could these be made no-ops?

Reply:
Not sure why I can't reply by the 2^31 issue, the kernel type is unsigned, but sure, as long as we can restrict to positive integer that's fine by me.  If you need metrics >=2^31 you can do the relevant patching.

Re no-op, will handle.  Normally I'd argue for "let it break so that someone knows something need to change", but in this case since it might well relate to something that the admin needs to gain access to the system ... I'm happy to concede.

Reply:
@paulusmack please check if you're happy with the change.

I took the opportunity to move options out of pppd/options.c and into the NCPs where they belong at the same time (at least those that I touched).  I trust this will be in-order as it honestly makes more sense to have IPCP options reside in ipcp.c and IPv6CP options in ipv6cp.c.

I did (from the initial PR) include some whitespace only changes into the PR.  If that's an issue I can rework to exclude that.  Didn't make the same mistake on other files (vim automatically cleans up trailing whitespace and a few others on my system).